layouts/shortcodes/cloakemail.html

```diff
@@ -1,4 +1,12 @@
-{{- $parts := split (index .Params 0) "@" -}}
+<!-- use mailto: if href is not set -->
+{{- $href := "mailto" -}}
+{{- if (.Get "href") -}}
+  {{- $href = (.Get "href") -}}
+{{ end }}
+<!-- create id for the element based on address, href and random number to make id unique -->
```

Comment:
I'd like to merge the "unique id" part. Is it possible to drop the "href" part for this PR (so you get credit for it in the commit history)?

Otherwise, I'll do the copy/paste myself. Thank you.

Reply:
Sure, please take whatever elements you like. The bad thing about introducing href is the missing backwards compatibility since the current code works with indexes, not parameters.

I'm not sure how I could edit this PR without erasing my current version which I use as a submodule in my website repo.

Reply:
> The bad thing about introducing href is the missing backwards compatibility since the current code works with indexes, not parameters.

Yes that's exactly my concern. I think there's a solution: IIRC one can detect if a shortcode has params, and otherwise use indexes.

Reply:
Found:
```go
{{ if .IsNamedParams }}
    "code for named parameters"
{{ else }}
    "code for positional parameters"
{{ end }}
```

Reply:
> I'm not sure how I could edit this PR without erasing my current version which I use as a submodule in my website repo.

 I think this is not possible :-(. I think I'll make a new version. When ready, would you agree to have a look at it?